vlad_task2/include/DynamicArray.hpp

```diff
@@ -3,28 +3,119 @@
 
 #include <string>
 #include <vector>
+#include <functional>
+#include <algorithm>
 
+template <typename T>
 class DynamicArray
 {
 public:
-    DynamicArray();
-    DynamicArray(const DynamicArray &other);
-    DynamicArray(const std::vector<std::string> &container);
-    DynamicArray(DynamicArray &&source);
-    DynamicArray &operator=(const DynamicArray &other);
-    DynamicArray &operator=(DynamicArray &&source);
-    ~DynamicArray();
-    
+    DynamicArray(): m_dynamicArray(nullptr), m_size(0)
+    {
+    }
+
+    DynamicArray(const DynamicArray<T> &other)
+    {
```

Comment:
Please implement the methods outside the class declaration in order to separate the declaration\
from the implementation.